statsmodels/regression/_prediction.py

```diff
@@ -166,10 +166,16 @@ def get_prediction(self, exog=None, transform=True, weights=None,
     var_pred_mean = (exog * np.dot(covb, exog.T).T).sum(1)
 
     # TODO: check that we have correct scale, Refactor scale #???
-    var_resid = self.scale / weights # self.mse_resid / weights
-    # special case for now:
-    if self.cov_type == 'fixed scale':
-        var_resid = self.cov_kwds['scale'] / weights
+    if weights is not None:
+        var_resid = self.scale / weights  # self.mse_resid / weights
+        # special case for now:
+        if self.cov_type == 'fixed scale':
+            var_resid = self.cov_kwds['scale'] / weights
+    else:
+        var_resid = self.scale
+        # special case for now:
+        if self.cov_type == 'fixed scale':
+            var_resid = self.cov_kwds['scale']
```

Comment:
there would be less code repetition with splitting the two ifs
`if self.cov_type == 'fixed scale'` without weights
and then
`if weights is not None` 
divide by weights

(based on the comment, the first `if` for fixed-scale should become redundant if we are sure self.scale is correctly `fixed`. IIRC self.scale is still the scale from the data and not what is used for the cov_params.)